images/salt-master/Dockerfile

```diff
@@ -15,9 +15,19 @@ gpgkey=https://repo.saltstack.com/yum/redhat/\$releasever/\$basearch/archive/%s/
  && rpm --import https://repo.saltstack.com/yum/redhat/7/x86_64/archive/${SALT_VERSION}/SALTSTACK-GPG-KEY.pub \
  && yum clean expire-cache \
  && yum install -y epel-release \
- && yum install -y python2-kubernetes salt-master salt-api salt-ssh openssh-clients \
+ && yum install -y salt-master salt-api salt-ssh openssh-clients \
  && yum install -y python-pip \
  && pip install pip==20.0.2 \
+ # Start of deps for python-kubernetes
+ # Note: Will replace pip install kubernetes & its deps below with a built rpm package
+ # Docker build will die here because we have PyYAML==3.11 installed
+ # ERROR: Cannot uninstall 'PyYAML'. It is a distutils installed project and
+ # thus we cannot accurately determine which files belong to it which would lead to only a partial uninstall
+ # Since kubernetes-client requires PyYAML==3.12 ipaddress==1.0.17 we can ignore the installed versions using:
+ && pip install setuptools==40.3.0 \
```

Comment:
Install-from-PyPI/pip is against the container building guidelines we tend to use. This should be packaged, then installed... We basically don't want to put any 'random' files (i.e., files from which the origin can't be retrieved from `rpm -qf ...` or similar) in either the container images we ship, or when installed on a machine, except for configuration files in `/etc` when applicable.

Reply:
We discussed this offline, and decided to provide a first version using pip, open a debt ticket to create some RPMs, so that we can close the addons upgrade tasks, currently blocked by this too old version of the client.

We are also installing `etcd3` with pip, so this one would require us to do some packaging as well. And there's the Python 3 dependencies... This will take some time, and while I agree we should enforce the guidelines you mentioned ASAP, I'd advocate we don't block other tasks while we are setting everything up.

(However, the comment in the Dockerfile could use a `TODO:` instead of just a `Note:` @Ebaneck, best would be to even have the issue number in there)

Reply:
We now have a dedicated issue here https://github.com/scality/metalk8s/issues/2454 to track the packaging of the k8s client